### ISSUE TITLE
Rule proposal: `no-date-parsing`

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2435]
### Description

`Date.parse` does not behave consistently across platforms

There is already an existing plugin for this https://github.com/amzn/eslint-plugin-no-date-parsing

But it requires typescrpt parser and does not support eslint 9

### Fail

```js
new Date('2024-08-16');

Date.parse('2024-08-16')
```

### Pass

```js
import { parseISO } from 'date-fns';

const date = parseISO('2024-08-16');
```

### Proposed rule name

no-date-parsing

**Comment by fisker** [comment]
Good idea, but I'm not sure if we can do it right without type information, we should only report strings.

**Comment by sindresorhus** [comment]
Ideally, it should be `Temporal.PlainDate.from('2024-08-16');`, but it's not available yet.

**Comment by geoffswift** [comment]
You could consider to autofix for a `YYYY-MM-DD` string like so...

```javascript
// Bad
new Date('2000-12-25');
```

```javascript
// Good
Date.UTC(2000, 11, 25);
```

Rules of the game here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format

**Comment by sindresorhus** [comment]
> Date.UTC(2000, 11, 25);

This is bad in other ways. The fact that month is zero-indexed is a huge footgun.

**Comment by sindresorhus** [comment]
I think it's better to wait for `Temporal`. It's coming: https://developer.mozilla.org/en-US/blog/javascript-temporal-is-coming/

**Comment by geoffswift** [comment]
`Temporal` is a good idea.  Part of the point I was making is string literals can at least be handled, without worrying about type infererences.  I think that's still sound at least.

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1315
